src/VecSim/utils/vec_utils.cpp

```diff
@@ -47,9 +47,9 @@ int cmpVecSimQueryResultByScore(const VecSimQueryResult *res1, const VecSimQuery
 }
 
 void float_vector_normalize(float *x, size_t dim) {
-    float sum = 0;
+    double sum = 0;
```

Comment:
why the change?

Reply:
when the floats in the vector are big, the sum is quickly becomes `inf`, and we turn the vector to \overline{0}
this way we can get more precise with the calculation of the norm